### DISCUSSION TITLE
How to debug the metapipeline?

### DISCUSSION THREAD

**Discussion by maotian06** [uclahs-cds/metapipeline-DNA#50] [Q&A] [answered]
@yashpatel6 @Alfredo-Enrique 
Hi, I was running my ACCA WXS samples using the metapipeline-DNA v1.0.0. After 2h 58m 3s, the pipeline failed.
Here is the log: `/hot/user/maotian/git/project-ACCA-NotchWnt/pipeline/metapipeline/log/meta-ACCA-1.log`
According to the log, my `align-DNA` failed, but when I checked the output folder's bam files. BAM files exist:
```
ls -lth /hot/project/disease/HeadNeckTumor/ACCA-000092-NotchWntMyc/output/metapipeline-DNA-1.0.0/ACCA-000092/main_workflow/output/align-DNA-8.0.0/ITACIMFS000001-T001-P01-P/BWA-MEM2-2.2.1/output/ITACIMFS000001-T001-P01-P.bam
ls -lth /hot/project/disease/HeadNeckTumor/ACCA-000092-NotchWntMyc/output/metapipeline-DNA-1.0.0/ACCA-000092/main_workflow/output/align-DNA-8.0.0/ITACIMFS000001-N001-A01-P/BWA-MEM2-2.2.1/output/ITACIMFS000001-N001-A01-P.bam
```
Maybe the call-gSNP failed? Any suggestion to debug the meta-pipeline?

**Comment by yashpatel6** [comment, marked as answer]
So for debugging with the metapipeline, you'll want to look at the specific logs for the patients. You can find these by looking at the main log file:
![image](https://user-images.githubusercontent.com/86321070/182964175-134fc498-1a98-4166-b2e6-3fc6681f0500.png)

At the beginning, the workDir locations correspond to the workDir in the common work dir, so these logs are accessible by going to where the common work dir you specified is then going to the the specific process folder. I this case, it would be `<common work dir>/36/568ac1...`. In there you can find the specific logs for the pipelines. Looking at the `.command.log` in that directory, it's a call-mtSNV error caused by a bad `gmapdb` path since that directory was moved/updated (you'll want to use `/hot/ref/mitochondria_ref/gmapdb/gmapdb_2021-03-08/`)

**Reply by maotian06** [reply]
Thanks!!!
